iModelJsNodeAddon/api_package/ts/src/test/DgnDb.test.ts

```diff
@@ -269,14 +266,38 @@ describe("basic tests", () => {
     let bisProps = db.getSchemaProps("BisCore");
     assert.isTrue(bisProps.version === "01.00.00");
     const schemaPath = path.join(iModelJsNative.DgnDb.getAssetsDir(), "ECSchemas/Domain/PresentationRules.ecschema.xml");
-    const result = db.importSchemas([schemaPath], { schemaLockHeld: false });
-    assert.isTrue(result === DbResult.BE_SQLITE_OK);
+    db.importSchemas([schemaPath], { schemaLockHeld: false });
 
     db.getSchemaProps("PresentationRules");
     bisProps = db.getSchemaProps("BisCore");
     assert.isTrue(bisProps.version >= "01.00.15"); // PR references 01.00.15+, so importing PR will cause it to upgrade.
   });
 
+  it("testSchemaImportNoAdditionalRootEntityClasses", () => {
+    const writeDbFileName = copyFile("noAdditionalRootEntityClasses.bim", dbFileName);
+    // Without ProfileOptions.Upgrade, we get: Error | ECDb | Failed to import schema 'BisCore.01.00.15'. Current ECDb profile version (4.0.0.1) only support schemas with EC version < 3.2. ECDb profile version upgrade is required to import schemas with EC Version >= 3.2.
+    const db = openDgnDb(writeDbFileName, { profile: ProfileOptions.Upgrade, schemaLockHeld: false });
+    assert.isTrue(db !== undefined);
+    const bisProps = db.getSchemaProps("BisCore");
+    assert.isTrue(bisProps.version === "01.00.00");
+
+    const schema = `<?xml version="1.0" encoding="utf-8" ?>
+    <ECSchema schemaName="TestSchema" alias="ts" version="1.0.0" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.3.2">
+        <ECSchemaReference name='ECDbMap' version='02.00.03' alias='ecdbmap' />
+        <ECEntityClass typeName="NewRootClass" modifier="Abstract">
+            <ECProperty propertyName="Name" typeName="string" />
+        </ECEntityClass>
+    </ECSchema>`;
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    const BE_SQLITE_ERROR_SchemaUpgradeFailed = (DbResult.BE_SQLITE_IOERR | 19 << 24);
+    try {
+      db.importXmlSchemas([schema], { schemaLockHeld: false });
+    } catch (error: any) {
+      assert.equal(error.errorNumber, BE_SQLITE_ERROR_SchemaUpgradeFailed);

```

Comment:
If the importXmlSchemas() function does not throw an exception, the test will still pass. It is expected that the function should throw an exception, in addition to the specific error message anticipated.

Reply:
Updated the test